src/commands/importFunction/importFunctionApp.ts

```diff
@@ -0,0 +1,405 @@
+            resOperations.push(nOperation);
+        } else {
+            resOperations.push(operation);
+        }
+    }
+    return resOperations;
+}
+
+function genUniqueOperationName(opName: string, existingOperations: string[]): number {
+    let cnt = 0;
+    let curName = opName;
+    while (existingOperations.includes(curName)) {
+        cnt++;
+        curName = opName.concat("-", cnt.toString());
+    }
+    return cnt;
+}
+
+async function getAllOperations(node: ApiTreeItem): Promise<string[]> {
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Get all operations from Api ${node.root.apiName}...`));
+    const operations: OperationCollection = await node.root.client.apiOperation.listByApi(node.root.resourceGroupName, node.root.serviceName, node.root.apiName);
+    const operationsNames: string[] = [];
+    operations.forEach(ele => {
+        operationsNames.push(String(ele.name));
+    });
+    return operationsNames;
+}
+
+async function getPickedFunctions(node: ApiTreeItem | ApisTreeItem, baseUrl: string): Promise<{
+    label: string;
+    func: IFunctionContract;
+}[]> {
+    const allFunctions = await listAllFunctions(baseUrl, node.root.credentials);
+    const importableFuncs = await filterFunctions(allFunctions);
+
+    // Pick functions to import
+    ext.outputChannel.appendLine(localize("importFunctionApp", "Get functions to import...."));
+    return await ext.ui.showQuickPick(importableFuncs.map((s) => { return { label: String(s.properties.name), func: s }; }), { canPickMany: true });
+}
+
+async function getFunctionApp(client: WebSiteManagementClient): Promise<{
+    label: string;
+    site: Site;
+}> {
+    // Get all webapps
+    ext.outputChannel.appendLine(localize("importFunctionApp", "Get function apps..."));
+    const allWebApps: WebAppCollection = await client.webApps.list();
+    const functionAppNames = allWebApps.filter((ele) => !!ele.kind && ele.kind.includes('functionapp'));
+
+    // Pick function app
+    return await ext.ui.showQuickPick(functionAppNames.map((s) => { return { label: String(s.name), site: s }; }), { canPickMany: false });
+}
+
+// Create policy for imported operation
+function createApiOperationXmlPolicy(backendId: string): string {
+    const operationPolicy = [{
+        policies: [
+            {
+                inbound: [
+                    { base: null },
+                    {
+                        "set-backend-service": [
+                            {
+                                _attr: {
+                                    id: "apim-generated-policy",
+                                    "backend-id": backendId
+                                }
+                            }
+                        ]
+                    }
+                ]
+            },
+            { backend: [{ base: null }] },
+            { outbound: [{ base: null }] },
+            { "on-error": [{ base: null }] }
+        ]
+    }];
+
+    return xml(operationPolicy);
```

Comment:
Not sure if we need this package for just this functionality

Reply:
yep, I tried several approaches, otherwise operation creation will fail with invalid policy content. I check how "serverless extension" deals with this